cmd/snap-confine/snap-confine.c

```diff
@@ -509,16 +542,58 @@ int main(int argc, char **argv)
 	return 1;
 }
 
-static void enter_classic_execution_environment(void)
+static void enter_classic_execution_environment(const sc_invocation * inv)
 {
 	/* 'classic confinement' is designed to run without the sandbox inside the
 	 * shared namespace. Specifically:
-	 * - snap-confine skips using the snap-specific mount namespace
+	 * - snap-confine skips using the snap-specific, private, mount namespace
 	 * - snap-confine skips using device cgroups
 	 * - snapd sets up a lenient AppArmor profile for snap-confine to use
 	 * - snapd sets up a lenient seccomp profile for snap-confine to use
 	 */
+
 	debug("skipping sandbox setup, classic confinement in use");
+
+	if (!sc_feature_enabled(SC_FEATURE_PARALLEL_INSTANCES)) {
+		return;
+	}
+
+	/* all of the following code is experimental and part of parallel instances
+	 * of classic snaps support */
+
+	debug
+	    ("(experimental) unsharing the mount namespace (per-classic-snap)");
+
+	/* Construct a mount namespace where the snap instance directories are
+	 * visible under the regular snap name. In order to do that we will:
+	 *
+	 * - convert SNAP_MOUNT_DIR into a mount point (global init)
+	 * - convert /var/snap into a mount point (global init)
+	 * - always create a new mount namespace
+	 * - for parallel instances of classic snaps only:
```

Comment:
This `for parallel instances of classic snaps only` is a bit out of date and implies we are doing something else with classic today in this function, but we are not due to logic elsewhere. This will be correct in the future though if/when we update the mount namespace for classic. Feel free to adjust the comment if you are doing other commits. Not a blocker.